src/components/Graph/Graph.jsx

```diff
@@ -1,119 +1,123 @@
 import React, { Component } from 'react';
 import { Line } from 'react-chartjs-2';
-import PropTypes from 'prop-types';
-import './Graph.css';
+import "./Graph.css";
 
-import loading from '../../img/loading.svg';
+import loading from "../../img/loading.svg";
 
 const offWhite = 'rgb(247, 248, 249)';
 const gray = '#929292';
 
 class Graph extends Component {
-  data = {
-    labels: this.props.labels,
-    datasets: [
-      {
-        fill: false,
-        borderColor: offWhite,
-        borderCapStyle: 'butt',
-        borderJoinStyle: 'miter',
-        borderWidth: 1,
-        pointRadius: 3,
-        pointBorderColor: offWhite,
-        pointBackgroundColor: offWhite,
-        pointHoverRadius: 10,
-        pointHoverBackgroundColor: offWhite,
-        pointHoverBorderWidth: 0,
-        pointHitRadius: 30,
-        lineTension: 0.4,
-        data: this.props.data,
-      },
-    ],
-  };
+   constructor(props) {
+      super(props);
+
+      this.state = {
+         data: this.getData()
+      }
+   }
+
+   componentDidUpdate(prevProps) {
+      if (prevProps.labels !== this.props.labels) {
+         this.setState({data: this.getData()});
+      }
+   }
+
+   getData() {
+      return {
+         labels: this.props.labels,
+         datasets: [
+            {
+               fill: false,
+               lineTension: 0,
+               borderColor: offWhite,
+               borderCapStyle: 'butt',
+               borderJoinStyle: 'miter',
+               borderWidth: 1,
+               pointRadius: 0,
+               pointBorderColor: offWhite,
+               pointBackgroundColor: offWhite,
+               pointHoverRadius: 3,
+               pointHoverBackgroundColor: offWhite,
+               pointHoverBorderWidth: 0,
+               pointHitRadius: 15,
+               lineTension: .4,
+               data: this.props.data
+            }
+         ]
+      }
+   }
+
+   loadingAnimation = (
+      <div className="Graph-loading-animation-wrapper">
+         <div className="Graph-loading-animation">
+            <img src={loading} alt="Loading" />
+         </div>
+      </div>
+   )
 
-  // Options for the chart
-  options = {
-    responsive: true,
-    maintainAspectRatio: false,
-    legend: {
-      display: false,
-    },
-    showAllTooltips: true,
-    tooltips: {
-      custom(tooltip) {
-        if (!tooltip) return;
-        // disable displaying the color box;
-        tooltip.displayColors = false;
+   // Options for the chart
+   options = {
```

Comment:
To keep it consistent I think it's a good idea to wrap this in a method as well something like getOptionsConfig.

Reply:
👍